## <b> Since 2008, guests and hosts have used Airbnb to expand on traveling possibilities and present a more unique, personalized way of experiencing the world. Today, Airbnb became one of a kind service that is used and recognized by the whole world. Data analysis on millions of listings provided through Airbnb is a crucial factor for the company. These millions of listings generate a lot of data - data that can be analyzed and used for security, business decisions, understanding of customers' and providers' (hosts) behavior and performance on the platform, guiding marketing initiatives, implementation of innovative additional services and much more. </b>

## <b>This dataset has around 49,000 observations in it with 16 columns and it is a mix between categorical and numeric values. </b>

## <b> Explore and analyze the data to discover key understandings (not limited to these) such as : 
* What can we learn about different hosts and areas?
* What can we learn from predictions? (ex: locations, prices, reviews, etc)
* Which hosts are the busiest and why?
* Is there any noticeable difference of traffic among different areas and what could be the reason for it? </b>

##**Importing Libraries**

In [1]:
#importing library
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline
import plotly.express as px
import warnings  
warnings.filterwarnings(action='ignore')

##**Loading Dataset**

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
#Reading CSV file
airbnb_data = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Project/Airbnb_Bookings_Analysis_Rajesh_Kumar_Patel-Capstone_Project_1/Airbnb NYC 2019.csv')

##**Data Exploration**

In [3]:
#how many rows and columns
airbnb_data.shape

(48895, 16)

48895 numbers of row and 16 columns present in our data

In [4]:
#summary of data
airbnb_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

In [5]:
#information about the data
airbnb_data.head(10)

id  ... availability_365
0  2539  ...              365
1  2595  ...              355
2  3647  ...              365
3  3831  ...              194
4  5022  ...                0
5  5099  ...              129
6  5121  ...                0
7  5178  ...              220
8  5203  ...                0
9  5238  ...              188

[10 rows x 16 columns]

In [6]:
#checking data types
airbnb_data.dtypes

id                                  int64
name                               object
host_id                             int64
host_name                          object
neighbourhood_group                object
neighbourhood                      object
latitude                          float64
longitude                         float64
room_type                          object
price                               int64
minimum_nights                      int64
number_of_reviews                   int64
last_review                        object
reviews_per_month                 float64
calculated_host_listings_count      int64
availability_365                    int64
dtype: object

Variable types

in our dataset have 10 Numeric feature and 6 Categorical feature

###**Feature information in details**

*  **id**  unique for every observation
* **name** name of properties
* **host_id** id for host
* **host_name** name of host
* **neighbourhood_group** it contains 5 Area ['Brooklyn', 'Manhattan', 'Queens', 'Staten Island', 'Bronx']
* **neighbourhood** it contains 221 locality names
* **latitude** it stores latitude
* **longitude** it stores longitude
* **room_type** it contains 3 unique room type ['Private room', 'Entire home/apt', 'Shared room']
* **price**  for property
* **minimum_nights** you have to book how many days
* **number_of_reviews** numbers of review in particular room
* **last_review** last_review date
* **reviews_per_month**  numbers of review per month
* **calculated_host_listings_count** Total number of listings owned by the host
* **availability_365** The number of days for which a particular host is available in a year


